fastlane/lib/fastlane/actions/github_api.rb

```diff
@@ -0,0 +1,266 @@
+                                           UI.user_error!("Unrecognised HTTP method")
+                                         end
+                                       end),
+          FastlaneCore::ConfigItem.new(key: :body,
+                                       env_name: "FL_GITHUB_API_REQUEST_BODY",
+                                       description: "The request body in JSON or hash format",
+                                       is_string: false,
+                                       default_value: {},
+                                       optional: true),
+          FastlaneCore::ConfigItem.new(key: :raw_body,
+                                       env_name: "FL_GITHUB_API_REQUEST_RAW_BODY",
+                                       description: "The request body taken vertabim instead of as JSON, useful for file uploads",
+                                       is_string: true,
+                                       optional: true),
+          FastlaneCore::ConfigItem.new(key: :path,
+                                       env_name: "FL_GITHUB_API_PATH",
+                                       description: "The endpoint path. e.g. '/repos/:owner/:repo/readme'",
+                                       optional: true),
+          FastlaneCore::ConfigItem.new(key: :url,
+                                       env_name: "FL_GITHUB_API_URL",
+                                       description: "The complete full url - used instead of path. e.g. 'https://uploads.github.com/repos/fastlane...'",
+                                       optional: true,
+                                       verify_block: proc do |value|
+                                         UI.user_error!("Please include the protocol in the url, e.g. https://uploads.github.com") unless value.include? "//"
+                                       end),
+          FastlaneCore::ConfigItem.new(key: :errors,
+                                       description: "Optional error handling hash based on status code, or pass '*' to handle all errors",
+                                       is_string: false,
+                                       default_value: {},
+                                       optional: true),
+          FastlaneCore::ConfigItem.new(key: :headers,
+                                       description: "Optional headers to apply",
+                                       is_string: false,
+                                       default_value: {},
+                                       optional: true),
+          FastlaneCore::ConfigItem.new(key: :secure,
+                                     env_name: "FL_GITHUB_API_SECURE",
+                                     description: "Optionally disable secure requests (ssl_verify_peer)",
+                                     is_string: false,
+                                     default_value: true,
+                                     optional: true),
+          FastlaneCore::ConfigItem.new(key: :debug,
```

Comment:
As mentioned above, I think we should get rid of this option and just use our verbose logging for this.